### PR TITLE
Add Boost Development page

### LINKED ISSUE #1559: Development / Release schedule does not match redirect
the previous development page https://www.boost.org/development/index.html does not correspond to it's boost.io redirect
https://www.boost.io/doc/contributor-guide/index.html

if possible, this should be made into a corresponding page with the same or similar functionality and linked on the "Learn" page.

### PR DESCRIPTION
**Pull request by GregKaleka**
Fixes #1559 

Screenshot 2024-12-20 at 10 21 05 AM
Screenshot 2024-12-20 at 10 24 57 AM

### CODE REVIEW

File: templates/boost_development.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{% extends "base.html" %}
+
+{% block title %}Boost Development{% endblock %}
+
+{% block content_wrapper %}
+<div class="flex justify-center items-center my-4">
+    <h1 class="text-2xl">Development</h1>
+</div>
+<div class="container px-4 mb-8 mx-auto">
```
Comment (rbbeeston) [inline templates/boost_development.html:9]: can we add the 'card' class here to wrap the top part in a card, and have less of the grey background?

**Comment by rbbeeston** [comment]
A couple of minor changes:

1. let's match the height of the header font to 3xl and align it left
2. Wrap the content using the 'card' class as shown in the image.

image

**Comment by GregKaleka** [comment]
Better?

Screenshot 2024-12-23 at 2 47 49 PM

**Comment by GregKaleka** [comment]
Fixed width issue, and also noticed the widgets needed to wrap on mobile, so I added some responsive classes.

Screenshot 2024-12-23 at 3 26 16 PM
Screenshot 2024-12-23 at 3 26 30 PM

**Review by rbbeeston — APPROVED** [review]
LGTM!

**Pull request merged**